hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableTest.java

```diff
@@ -349,8 +351,70 @@ public void testListTables() throws TException, IOException {
     HIVE_METASTORE_EXTENSION.metastoreClient().dropTable(DB_NAME, hiveTableName);
   }
 
-  private org.apache.hadoop.hive.metastore.api.Table createHiveTable(String hiveTableName)
-      throws IOException {
+  @Test
```

Comment:
you could change this to a `@ParameterizedTest`, since the only difference between the two tests is the `TableType`. So the test would have a `TableType` parameter, where you pass `TableType.EXTERNAL_TABLE` and `TableType.VIRTUAL_VIEW` via `@EnumSource` (there are examples in the codebase on how to use parameterized tests with enums)